src/pages/ReimbursementAccount/ACHContractStep.js

```diff
@@ -115,56 +148,31 @@ class ACHContractStep extends React.Component {
     }
 
     /**
-     * Clear the error associated to inputKey if found and store the inputKey new value in the state.
-     *
-     * @param {Integer} ownerIndex
-     * @param {Object} values
+     * @param {Object} values - object containing form input values
      */
-    clearErrorAndSetBeneficialOwnerValues(ownerIndex, values) {
-        this.setState((prevState) => {
-            const beneficialOwners = [...prevState.beneficialOwners];
-            beneficialOwners[ownerIndex] = {...beneficialOwners[ownerIndex], ...values};
-            BankAccounts.updateReimbursementAccountDraft({beneficialOwners});
-            return {beneficialOwners};
-        });
-
-        // Prepare inputKeys for clearing errors
-        const inputKeys = _.keys(values);
-
-        // dob field has multiple validations/errors, we are handling it temporarily like this.
-        if (_.contains(inputKeys, 'dob')) {
-            inputKeys.push('dobAge');
-        }
-        this.clearErrors(_.map(inputKeys, inputKey => `beneficialOwnersErrors.${ownerIndex}.${inputKey}`));
-    }
-
-    submit() {
-        if (!this.validate()) {
-            return;
-        }
-
+    submit(values) {
         const bankAccountID = lodashGet(store.getReimbursementAccountInSetup(), 'bankAccountID');
 
-        // If they did not select that there are other beneficial owners, then we need to clear out the array here. The
-        // reason we do it here is that if they filled out several beneficial owners, but then toggled the checkbox, we
-        // want the data to remain in the form so we don't lose the user input until they submit the form. This will
-        // prevent the data from being sent to the API
```

Comment:
Should this comment be removed? Or is it still valuable?

Reply:
Just double checking on why we are removing this comment.

Reply:
Ah ok, maybe we can update it so that it explains why we are setting the `beneficialOwners` to an empty array, but I think it's maybe obvious why.